tests/lib/external/snapd-testing-tools/tools/repack-kernel

```diff
@@ -0,0 +1,231 @@
+#!/bin/bash
+set -e
+
+if [ -n "$D" ]; then
+    set -x
+fi
+
+show_help() {
+    echo "usage: repack-kernel <command> <opts>"
+    echo ""
+    echo "handle extraction of the kernel snap to a workspace directory, and later"
+    echo "repacking it back to a snap."
+    echo ""
+    echo "   repack-kernel setup                        - setup system dependencies"
+    echo "   repack-kernel extract <snap-file> <target> - extract under <target> workspace tree"
+    echo "   repack-kernel prepare <target>             - prepare initramfs & kernel for repacking"
+    echo "   repack-kernel pack <target>                - pack the kernel"
+    echo "   repack-kernel cull-firmware <target>       - remove unnecessary firmware"
+    echo "   repack-kernel cull-modules <target>        - remove unnecessary modules"
+    echo ""
+}
+
+setup() {
+    if [ "$UID" != "0" ]; then
+        echo "run as root (only this command)"
+        exit 1
+    fi
+
+    # carries ubuntu-core-initframfs
+    add-apt-repository ppa:snappy-dev/image -y
+    apt update -y
+    
+    if ! dpkg -s ubuntu-core-initramfs >/dev/null 2>&1; then
+        if ! apt install ubuntu-core-initramfs -y; then
+            echo "ubuntu-core-initramfs was not available in deb repository!"
+            exit 1
+        fi
+    fi
+}
+
+get_kver() {
+    local kerneldir="$1"
+    (
+        cd "$kerneldir"
+        #shellcheck disable=SC2010
+        ls "config"-* | grep -Po 'config-\K.*'
```

Comment:
I think something like:
`find "$kerneldir" -maxdepth 1 -name "config"-* | grep -Po 'config-\K.*'`
would avoid the subshell and the shellcheck warning.

Reply:
Awesome, thanks for the suggestion!

Reply:
I just tested this and your suggestion works just fine and avoids the subshell/shellcheck warning